File: qt/gen_qlatin1stringmatcher.go
package qt

import (
	"unsafe"
)

type QLatin1StringMatcher struct {
	h          uintptr
	isSubclass bool
}

// NewQLatin1StringMatcher constructs a new QLatin1StringMatcher object.
func NewQLatin1StringMatcher() *QLatin1StringMatcher {
	ret := newQLatin1StringMatcher(QLatin1StringMatcher_new())
	ret.isSubclass = true
	return ret
}

func (this *QLatin1StringMatcher) SetCaseSensitivity(cs CaseSensitivity) {
	QLatin1StringMatcher_SetCaseSensitivity(this.h, (int)(cs))
}

func (this *QLatin1StringMatcher) CaseSensitivity() CaseSensitivity {
	return (CaseSensitivity)(QLatin1StringMatcher_CaseSensitivity(this.h))
}

File: qt/gen_qlatin1stringmatcher.cpp
// +build ignore

#include <QLatin1StringMatcher>
#include <qlatin1stringmatcher.h>
#include "gen_qlatin1stringmatcher.h"

#ifndef _Bool
#define _Bool bool
#endif

QLatin1StringMatcher* QLatin1StringMatcher_new() {
	return new QLatin1StringMatcher();
}

void QLatin1StringMatcher_SetCaseSensitivity(QLatin1StringMatcher* self, int cs) {
	self->setCaseSensitivity(static_cast<Qt::CaseSensitivity>(cs));
}

int QLatin1StringMatcher_CaseSensitivity(const QLatin1StringMatcher* self) {
	Qt::CaseSensitivity _ret = self->caseSensitivity();
	return static_cast<int>(_ret);
}

void QLatin1StringMatcher_Delete(QLatin1StringMatcher* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QLatin1StringMatcher*>( self );
	} else {
		delete self;
	}
}

